### PR TITLE
[query] Add missing "status" field to PromQL error responses

### PR DESCRIPTION
**Pull request by vpranckaitis**
**What this PR does / why we need it**:

Adds missing status field to PromQL error responses + related tests

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
NONE
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note
NONE
```

### CODE REVIEW

File: src/x/net/http/errors.go [hunk 1]
```diff
@@ -68,7 +68,8 @@ func (e errorWithCode) Code() int {
 
 // ErrorResponse is a generic response for an HTTP error.
 type ErrorResponse struct {
-	Error string `json:"error"`
+	Status string `json:"status"`
+	Error  string `json:"error"`
 }
```
Comment (vpranckaitis) [inline src/x/net/http/errors.go:73]: https://prometheus.io/docs/prometheus/latest/querying/api/#format-overview

**Review by robskillington — APPROVED** [review]
LGTM

**Comment by vpranckaitis** [comment]
@ChrisChinchilla adding you for review since there's a tiny OpenAPI spec change https://github.com/m3db/m3/pull/2933/files#diff-6c9ead05879801460adf1d4c82cbd65be2bb82754876a460b77ba63900237d00R859-R860

**Review by ChrisChinchilla — APPROVED** [review]
Looks fine to me 👍

**Pull request merged**